crates/oxc_semantic/src/builder.rs

```diff
@@ -1230,7 +1232,10 @@ impl<'a> Visit<'a> for SemanticBuilder<'a> {
         /* cfg */
 
         let ret_kind = if let Some(arg) = &stmt.argument {
+            let in_return = self.is_in_return;
+            self.is_in_return = true;
             self.visit_expression(arg);
+            self.is_in_return = in_return;
```

Comment:
This may cause a bit of difficulty in maintaining. We should avoid writing any logic in the visit function.